logstash-core/lib/logstash/agent.rb

```diff
@@ -262,4 +268,8 @@ def clean_state?
   def reset_collector
     LogStash::Instrument::Collector.instance.clear
   end
+
+  def setting(key)
```

Comment:
+1 on using an instance of Settings passed in (@settings) instead of a global LogStash::SETTINGS for reasons you stated (easier testing)